### PR TITLE
[stable/kube-state-metrics] add support for useExistingRole

### PR DESCRIPTION
**Pull request by KlavsKlavsen**
This is a repost of PR https://github.com/helm/charts/pull/23059

This adds support for running on kubernetes setups (such as openshift) where you only have access to your own namespaces.

I've gotten this support merged in both Prometheus and Grafana charts and
I'd really like to get it in this last chart I use for our Prometheus setup ;)

### CODE REVIEW

File: stable/kube-state-metrics/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ keywords:
 - monitoring
 - prometheus
 - kubernetes
-version: 2.9.1
+version: 2.10.1
```
Comment (mrueg) [inline stable/kube-state-metrics/Chart.yaml]: Please use 2.10.0 here

**Comment by mrueg** [comment]
@tariq1890 can you take a look at this one as well? Thanks!

File: stable/kube-state-metrics/templates/role.yaml [hunk 2]
```diff
@@ -0,0 +1,184 @@
+{{- if and (eq $.Values.rbac.create true) (.Values.namespace) (eq .Values.rbac.useClusterRole false) (not .Values.rbac.useExistingRole)  -}}
```
Comment (mrueg) [inline stable/kube-state-metrics/templates/role.yaml]: Could the clusterrole and this new role be merged, so we can apply DRY to the rbac permissions?

Comment (KlavsKlavsen) [reply stable/kube-state-metrics/templates/role.yaml]: I've made an attempt at that.. it generates the same yaml for my testcase (multiple namespaces and useClusterrole false)..

File: stable/kube-state-metrics/Chart.yaml (hunk 1 above)
Comment (KlavsKlavsen) [reply stable/kube-state-metrics/Chart.yaml]: done

**Comment by mrueg** [comment]
@KlavsKlavsen can you make sure the title of the PR starts with [stable/kube-state-metrics]?

File: stable/kube-state-metrics/templates/stsdiscovery-role.yaml [hunk 3]
```diff
@@ -19,11 +19,9 @@ rules:
 - apiGroups:
   - apps
   resourceNames:
-  - {{ template "kube-state-metrics.fullname" . }}
+  - kube-state-metrics
```
Comment (mrueg) [inline stable/kube-state-metrics/templates/stsdiscovery-role.yaml]: why is this change needed?

File: stable/kube-state-metrics/templates/role.yaml [hunk 4]
```diff
@@ -1,180 +1,194 @@
...
 {{ end -}}
-{{ if .Values.collectors.verticalpodautoscalers }}
+{{ if $.Values.collectors.verticalpodautoscalers }}
 - apiGroups: ["autoscaling.k8s.io"]
   resources:
     - verticalpodautoscalers
   verbs: ["list", "watch"]
 {{ end -}}
+{{- if eq .Values.rbac.useClusterRole false }}
+{{-   end -}}
```
Comment (mrueg) [inline stable/kube-state-metrics/templates/role.yaml]: seems like here is an additional empty space

File: stable/kube-state-metrics/templates/role.yaml [hunk 5]
```diff
@@ -1,180 +1,194 @@
...
   verbs: ["list", "watch"]
 {{ end -}}
-{{ if .Values.collectors.verticalpodautoscalers }}
+{{ if $.Values.collectors.verticalpodautoscalers }}
 - apiGroups: ["autoscaling.k8s.io"]
   resources:
     - verticalpodautoscalers
   verbs: ["list", "watch"]
 {{ end -}}
+{{- if eq .Values.rbac.useClusterRole false }}
```
Comment (mrueg) [inline stable/kube-state-metrics/templates/role.yaml]: is this one correct here?

File: stable/kube-state-metrics/templates/stsdiscovery-role.yaml (hunk 3 above)
Comment (KlavsKlavsen) [reply stable/kube-state-metrics/templates/stsdiscovery-role.yaml]: It really was't - bad merge on my part. I've fixed it up - also found a few others. There should have been no changes to stsdiscovery-role.yaml :(

File: stable/kube-state-metrics/templates/role.yaml (hunk 5 above)
Comment (KlavsKlavsen) [reply stable/kube-state-metrics/templates/role.yaml]: no

**Comment by KlavsKlavsen** [comment]
I've merged latest upstream again.

**Comment by KlavsKlavsen** [comment]
@scottrigby @mrueg Have I not resolved all issues? Whats outstanding? what can be improved ?

**Comment by scottrigby** [comment]
@KlavsKlavsen this project is now read-only. Please see:
- Announcement https://helm.sh/blog/helm-2-becomes-unsupported/
- For chart relocation info: #21103

**Comment by scottrigby** [comment]
Specifically see https://github.com/kubernetes/kube-state-metrics/pull/1237 for current status of relocating stable/kube-state-metrics. We are waiting on kubernetes legal for final approval for that PR.￼

**Comment by KlavsKlavsen** [comment]
1325 was merged
